pandas/core/ops/array_ops.py

```diff
@@ -291,6 +294,7 @@ def na_logical_op(x: np.ndarray, y, op):
             y = ensure_object(y)
             result = libops.vec_binop(x.ravel(), y.ravel(), op)
         else:
+            x = ensure_object(x)
```

Comment:
if we're doing this both here and on L293, then might as well do it just once after L289

Do we have cases that get here with x not object dtype?